test/unit/utils.py

```diff
@@ -15,13 +15,41 @@
 
 """A collection of functions that helps running unit tests"""
 
+import eventlet.wsgi
 import functools
+import random
 import re
 import unittest
 
 import nose.plugins.skip
 
 
+class WSGIServer(object):
+    """Start a WSGI Web server."""
+    def __init__(self, application):
+        self.application = application
+        self.thread = None
+
+        self.server = eventlet.listen(('127.0.0.1', 0))
```

Comment:
I'd move these 2 lines into `start`, so the constructor itself is free of side-effects. Who's supposed to close this socket otherwise (except for the GC)?